tutorials/llm/2_prompt_usage.py

```diff
@@ -0,0 +1,158 @@
+# %% [markdown]
+"""
+# LLM: 2. Prompt Usage
+
+Prompting is an essential step in using LLMs and Chatsky provides you with a simple way of using multiple prompts throughout your script.
+The main idea behind that is the hierarchy of prompts. It can be portrayed as follows:
+```
+SYSTEM: SYSTEM_PROMPT
+SYSTEM: NODE_PROMPT
+SYSTEM: GLOBAL_PROMPT
+SYSTEM: LOCAL_PROMPT
+
+# history `n` turns
+HUMAN: req
+AI: resp
+
+SYSTEM: RESPONSE_PROMPT
+HUMAN: CURRENT_REQUEST
+```
+
+This way you can specify a certain prompt for each flow or node to alter the model's behavior.
+"""
+
+# %pip install chatsky[llm]
+
+# %%
+from chatsky.core.message import Message
+from chatsky import (
+    TRANSITIONS,
+    RESPONSE,
+    GLOBAL,
+    LOCAL,
+    MISC,
+    Pipeline,
+    Transition as Tr,
+    conditions as cnd,
+    destinations as dst,
+)
+from chatsky.utils.testing import is_interactive_mode
+from chatsky.llm import LLM_API
+from chatsky.responses.llm import LLMResponse
+
+
+import os
+import re
+
+os.environ["OPENAI_API_KEY"] = "<TOKEN>"
+
+from langchain_openai import ChatOpenAI
+
+# %% [markdown]
+"""
+Let's create a simple script to demonstrate this. Note, that prompts should go the `MISC` field of the node. Inside `MISC` they can be stored under the `prompt`, `global_prompt` or `local_prompt` keys.
+Please note, that `MISC` is just a dictionary and its fields can be overwritten in any node, if given the same key. You can utilize this in your project, but below we present the intended way of using `MISC` for storing multiple prompts.
+"""
+
+# %%
+# this `system_prompt` will be always on the top of the history during models response
+model = LLM_API(
+    ChatOpenAI(model="gpt-3.5-turbo"),
+    system_prompt="You will represent different bank workers. Answer users' questions according to your role.",
+)
+llm_response = LLMResponse()
+
+toy_script = {
+    GLOBAL: {
+        MISC: {
+            # this prompt will be overwritten with every node with `prompt` key in it
+            "prompt": "Your role is a bank receptionist. Provide user with the information about our bank and the services we can offer.",
```

Comment:
Create prompt class:
```
class BasePrompt:
  prompt: AnyResponse
  priority: AnyPriority
```

Reply:
1. Prompt discovery from MISC 
	1. add regex for determining prompts: ".*_prompt|prompt_.*"; `LLMResponse(..., prompt_key_regex=) ` to allow different naming conventions (`..._openai_prompt`; `..._mistral_prompt`)
2. Default positions (configurable):
	1. `LLMResponse(..., default_positions={"system_prompt": 0, "history": 1, "node_prompts": 2, "call_prompt": 3, "last_response": 4})`:
		`BasePrompt(position=)` will be placed between `system_prompt` and `history`.
Configs can be set via `LLM_API` or `LLMResponse`.